File: memprint.py
import sys

def interpret(code):
    lines = code.split("\n")
    memory = 26

    for line in lines:
        tokens = list(line)

        try:
            if tokens[0] == "#":
                pass
            else:
                for token in tokens:
                    if token == "*":
                        memory += 5
                    elif token == "+":
                        memory += 1
                    elif token == "-":
                        memory -= 1
                    elif token == "!":
                        memory -= 5
                    elif token == "&":
                        if memory == 26:
                            print("a", end="")
                        elif memory == 25:
                            print("b", end="")
                        elif memory == 24:
                            print("c", end="")
                        elif memory == 23:
                            print("d", end="")
                        elif memory == 22:
                            print("e", end="")
                        elif memory == 21:
                            print("f", end="")
                        elif memory == 20:
                            print("g", end="")
                        elif memory == 19:
                            print("h", end="")
                        elif memory == 18:
                            print("i", end="")
                        elif memory == 17:
                            print("j", end="")
                        elif memory == 16:
                            print("k", end="")
                        elif memory == 15:
                            print("l", end="")
                        elif memory == 14:
                            print("m", end="")
                        elif memory == 13:
                            print("n", end="")
                        elif memory == 12:
                            print("o", end="")
                        elif memory == 11:
                            print("p", end="")
                        elif memory == 10:
                            print("q", end="")
                        elif memory == 9:
                            print("r", end="")
                        elif memory == 8:
                            print("s", end="")
                        elif memory == 7:
                            print("t", end="")
                        elif memory == 6:
                            print("u", end="")
                        elif memory == 5:
                            print("v", end="")
                        elif memory == 4:
                            print("w", end="")
                        elif memory == 3:
                            print("x", end="")
                        elif memory == 2:
                            print("y", end="")
                        elif memory == 1:
                            print("z", end="")
                        elif memory == 27:
                            print(" ", end="")
                        elif memory == 28:
                            print("\n", end="")
                        elif memory == 29:
                            print("!", end="")
                        elif memory == 30:
                            print("?", end="")
                        elif memory == 31:
                            print(",", end="")
                        elif memory == 32:
                            print(".", end="")
                        elif memory == 33:
                            print("+", end="")
                        elif memory == 34:
                            print("-", end="")
                        elif memory == 35:
                            print("*", end="")
                        elif memory == 36:
                            print("&", end="")
                        elif memory == 37:
                            print("{", end="")
                        elif memory == 38:
                            print("}", end="")
                        elif memory == 39:
                            print("%", end="")
                        elif memory == 40:
                            print("[", end="")
                        elif memory == 41:
                            print("]", end="")
                        elif memory == 42:
                            print("=", end="")
                        elif memory == 43:
                            print(";", end="")
                        elif memory == 44:
                            print(":", end="")
                        elif memory == 45:
                            print("#", end="")
                        elif memory == 46:
                            print("$", end="")
                        elif memory == 47:
                            print("\"", end="")
                        elif memory == 48:
                            print("'", end="")
                        elif memory == 49:
                            print("~", end="")
                        elif memory == 50:
                            print("0", end="")
                        elif memory == 51:
                            print("1", end="")
                        elif memory == 52:
                            print("2", end="")
                        elif memory == 53:
                            print("3", end="")
                        elif memory == 54:
                            print("4", end="")
                        elif memory == 55:
                            print("5", end="")
                        elif memory == 56:
                            print("6", end="")
                        elif memory == 57:
                            print("7", end="")
                        elif memory == 58:
                            print("8", end="")
                        elif memory == 59:
                            print("9", end="")
                        elif memory == 60:
                            print(">", end="")
                        elif memory == 61:
                            print("<", end="")
                    elif token == "%":
                        print(memory)
        except IndexError:
            print("", end="")

if __name__ == "__main__":
    if len(sys.argv) == 1:
        print(f"Usage: {sys.argv[0]} <file>")
    else:
        if sys.argv[1].endswith(".lemem"):
            with open(sys.argv[1], "r") as f:
                interpret(f.read())
        else:
            print("Use .lemem file extension")

File: memprint.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

void interpret(char* code) {
    char* line;
    char* token;
    int memory = 26;
    char inp[100] = "";

    line = strtok(code, "\n");
    while (line != NULL) {
        token = strtok(line, "");
        while (*token != '\0') {
            if (*token == '*')
                memory += 5;
            else if (*token == '+')
                memory += 1;
            else if (*token == '-')
                memory -= 1;
            else if (*token == '!')
                memory -= 5;
            else if (*token == '&') {
                switch (memory) {
                    case 26: printf("a"); break;
                    case 25: printf("b"); break;
                    case 24: printf("c"); break;
                    case 23: printf("d"); break;
                    case 22: printf("e"); break;
                    case 21: printf("f"); break;
                    case 20: printf("g"); break;
                    case 19: printf("h"); break;
                    case 18: printf("i"); break;
                    case 17: printf("j"); break;
                    case 16: printf("k"); break;
                    case 15: printf("l"); break;
                    case 14: printf("m"); break;
                    case 13: printf("n"); break;
                    case 12: printf("o"); break;
                    case 11: printf("p"); break;
                    case 10: printf("q"); break;
                    case 9: printf("r"); break;
                    case 8: printf("s"); break;
                    case 7: printf("t"); break;
                    case 6: printf("u"); break;
                    case 5: printf("v"); break;
                    case 4: printf("w"); break;
                    case 3: printf("x"); break;
                    case 2: printf("y"); break;
                    case 1: printf("z"); break;
                    case 27: printf(" "); break;
                    case 28: printf("\n"); break;
                    case 29: printf("!"); break;
                    case 30: printf("?"); break;
                    case 31: printf(","); break;
                    case 32: printf("."); break;
                    case 33: printf("+"); break;
                    case 34: printf("-"); break;
                    case 35: printf("*"); break;
                    case 36: printf("&"); break;
                    case 37: printf("{"); break;
                    case 38: printf("}"); break;
                    case 39: printf("\%"); break;
                    case 40: printf("["); break;
                    case 41: printf("]"); break;
                    case 42: printf("="); break;
                    case 43: printf(";"); break;
                    case 44: printf(":"); break;
                    case 45: printf("#"); break;
                    case 46: printf("$"); break;
                    case 47: printf("\""); break;
                    case 48: printf("'"); break;
                    case 49: printf("~"); break;
                    case 50: printf("0"); break;
                    case 51: printf("1"); break;
                    case 52: printf("2"); break;
                    case 53: printf("3"); break;
                    case 54: printf("4"); break;
                    case 55: printf("5"); break;
                    case 56: printf("6"); break;
                    case 57: printf("7"); break;
                    case 58: printf("8"); break;
                    case 59: printf("9"); break;
                    case 60: printf(">"); break;
                    case 61: printf("<"); break;
                    case 62: printf("("); break;
                    case 63: printf(")"); break;
                    case 64: printf("^"); break;
                }
            } else if (*token == '%') {
                printf("%d", memory);
            } else if (*token == '@') {
                scanf("%s", inp);
            } else if (*token == '.') {
                printf("%s", inp);
            }
            token++;
        }
        line = strtok(NULL, "\n");
    }
}

int main(int argc, char* argv[]) {
    char version[] = "1.1";
    if (argc == 1) {
        printf("memprint version: %s\n", version);
        printf("Usage: %s <file>\n", argv[0]);
    } else {
        if (strstr(argv[1], ".mempr") != NULL) {
            FILE* file = fopen(argv[1], "r");
            if (file != NULL) {
                fseek(file, 0, SEEK_END);
                long size = ftell(file);
                fseek(file, 0, SEEK_SET);
                char* buffer = (char*)malloc(size + 1);
                if (buffer != NULL) {
                    fread(buffer, 1, size, file);
                    buffer[size] = '\0';
                    interpret(buffer);
                    free(buffer);
                }
                fclose(file);
            }
        } else {
            printf("Use .mempr file extension\n");
        }
    }
    return 0;
}
